### PR TITLE
Set default template based on the available operating systems

### PR DESCRIPTION
**Pull request by maciaszczykm**
### What this PR does / why we need it
Previously Ubuntu was used as default even if it wasn't configured for the chosen datacenter. Right now, the datacenter configuration is taken under consideration.

### Which issue(s) this PR fixes
Fixes https://github.com/kubermatic/dashboard/issues/3207

### Release note

```release-note
Set default VM template based on the available operating systems.
```

### CODE REVIEW

File: src/app/node-data/basic/provider/vsphere/component.ts [hunk 1]
```diff
@@ -121,7 +121,8 @@ export class VSphereBasicNodeDataComponent extends BaseFormValidator implements
     }
   }
 
-  private _setDefaultTemplate(os: OperatingSystem): void {
+  private _setDefaultTemplate(os: OperatingSystem = undefined): void {
```
Comment (floreks) [inline src/app/node-data/basic/provider/vsphere/component.ts:124]: If this parameter is no longer used, I'd remove it.

Comment (maciaszczykm) [reply src/app/node-data/basic/provider/vsphere/component.ts:124]: It's used in line 108.

**Pull request merged**